### ISSUE TITLE
Provide a way to send Liberty Access Logs to OpenTelemetry

### ISSUE THREAD

**Issue by donbourne** [OpenLiberty/open-liberty#29228]
## Description

We need a way for users to be able to direct their Liberty access logs to OpenTelemetry.

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## Documents

When available, add links to required feature documents. Use "N/A" to mark particular documents which are not required by the feature.

- Externally raised requests for enhancements: 
  - [Aha](https://cloud-platform.ideas.aha.io/): Link to the Aha idea (also add a link to this issue in the Aha idea)
    - Feature owner adds label `Aha idea`
  - [Open Liberty Feature Request](https://github.com/OpenLiberty/open-liberty/issues/new?assignees=&labels=enhancement&template=feature_request.md&title=): Link to the OL GH issue (also add a link to this issue in the Feature Request GH issue)
    - Feature owner adds label `Requested feature`
- UFO: https://ibm.box.com/s/nq85gc51xe3qf15975o9s8dva7lr0y3c
- FTS: [Feature Test Summary for Provide a way to send Liberty Access Logs to OpenTelemetry](https://github.com/OpenLiberty/open-liberty/issues/31064)
- Beta Blog(s): [Providing Liberty access logs to OpenTelemetry using MicroProfile Telemetry 2.0](https://github.com/OpenLiberty/open-liberty/issues/30977)
- GA Blog: [Providing Liberty access logs to OpenTelemetry using MicroProfile Telemetry 2.0](https://github.com/OpenLiberty/open-liberty/issues/31481)

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## Process Overview

- [Prioritization](#prioritization)
- [Design](#design)
- [Implementation](#implementation)
- [Legal and Translation](#legal-and-translation)
- [Beta](#beta)
- [GA](#ga)
  - [Focal Point Approvals](#focal-point-approvals-complete-by-feature-complete-date)
- [Other Deliverables](#other-deliverables)

### General Instructions

The process steps occur roughly in the order as presented. Process steps occasionally overlap.

Each process step has a number of tasks which must be completed or must be marked as not applicable ("N/A").

Unless otherwise indicated, the tasks are the responsibility of the feature owner or a delegate of the feature owner.

If you need assistance, reach out to the [OpenLiberty/release-architect](https://github.com/orgs/OpenLiberty/teams/release-architect).

**Important: Labels are used to trigger particular steps and must be added as indicated.** 

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## **Prioritization** (Complete Before Development Starts)

The [OpenLiberty/chief-architect](https://github.com/orgs/OpenLiberty/teams/chief-architect) and area leads are responsible for prioritizing the features and determining which features are being actively worked on.

### **Prioritization**
- [ ] Feature owner adds label `Prioritization - Requested`
  - This puts the feature on the radar of the [OpenLiberty/chief-architect](https://github.com/orgs/OpenLiberty/teams/chief-architect) and [OpenLiberty/project-manager](https://github.com/orgs/OpenLiberty/teams/project-manager). They are responsible for querying for new features that need to be prioritized.
- [ ] [OpenLiberty/project-manager](https://github.com/orgs/OpenLiberty/teams/project-manager) adds feature to the "New" column of the [Open Liberty project board](https://github.com/orgs/OpenLiberty/projects/2)
- [ ] Priority assigned
  - Attend the Liberty Backlog Prioritization meeting
  - `Prioritization - Requested` label removed ([OpenLiberty/project-manager](https://github.com/orgs/OpenLiberty/teams/project-manager) or feature owner)

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## **Design** (Complete Before Development Starts)

Design preliminaries determine whether a formal design, which will be provided by an [Upcoming Feature Overview (UFO)](https://ibm.box.com/v/UFO-Template) document, must be created and reviewed.  A formal design is required if the feature requires any of the following: UI, Serviceability, SVT, Performance testing, or non-trivial documentation/ID.  Furthermore, each identified item places a blocking requirement on another team so it must be identified early in the process.  The feature owner may check-off the item if they know it doesn't apply, but otherwise they should work with the focal point to determine what work, if any, will be necessary and make them aware of it.

### **Design Preliminaries**
- [x] UI requirements identified, or N/A. (Feature owner and [UI focal point](https://github.com/orgs/OpenLiberty/teams/ui-approvers))
- [ ] Accessibility requirements identified, or N/A. (Feature owner and [Accessibility focal point](https://github.com/orgs/OpenLiberty/teams/accessibility-approvers))
- [ ] ID requirements identified, or N/A. (Feature owner and [ID focal point](https://github.com/orgs/OpenLiberty/teams/id-approvers))
   - Refer to [Documenting Open Liberty](https://github.com/OpenLiberty/open-liberty/wiki/Documenting-Open-Liberty).
   - Feature owner adds label `ID Required`, if non-trivial documentation needs to be created by the ID team.
   - ID adds label `ID Required - Trivial`, if no design will be performed and only trivial ID updates are needed.
- [ ] Serviceability requirements identified, or N/A. (Feature owner and [Serviceability focal point](https://github.com/orgs/OpenLiberty/teams/serviceability-approvers))
- [ ] SVT requirements identified, or N/A. (Feature owner and [SVT focal point](https://github.com/orgs/OpenLiberty/teams/svt-approvers))
- [ ] Performance testing requirements identified, or N/A. (Feature owner and [Performance focal point](https://github.com/orgs/OpenLiberty/teams/performance-approvers))

### **Design**
- [x] POC Design / UFO review requested.
  - Feature owner adds label `Design Review Request`
- [x] POC Design / UFO review scheduled.
  - Follow the instructions in POC-Forum repo
- [x] POC Design / UFO review completed.
- [x] POC / UFO Review follow-ons completed.
- [x] POC Design / UFO approval requested.
  - Feature owner adds label `Design Approval Request`
- [x] Design / UFO approved. ([OpenLiberty/chief-architect](https://github.com/orgs/OpenLiberty/teams/chief-architect)) or N/A
  - ([OpenLiberty/chief-architect](https://github.com/orgs/OpenLiberty/teams/chief-architect)) adds label `Design Approved`
  - Add the public link to the UFO in Box to the [Documents](#documents) section.
  - The UFO must always accurately reflect the final implementation of the feature. Any changes must be first approved. Afterwards, update the UFO by creating a copy of the original approved slide(s) at the end of the deck and prepend "OLD" to the title(s). A single updated copy of the slide(s) should take the original's place, and have its title(s) prepended with "UPDATED".

### **No Design**
- [ ] No Design requested.
  - Feature owner adds label `No Design Approval Request`
- [ ] No Design / No UFO approved. ([OpenLiberty/chief-architect](https://github.com/orgs/OpenLiberty/teams/chief-architect)) or N/A
  - Approver adds label `No Design Approved`
- [ ] Feature / Capability stabilization or discontinuation or N/A
  - Feature owner adds label `Product Management Approval Request` and notifies [OpenLiberty/product-management](https://github.com/orgs/OpenLiberty/teams/product-management)
  - Approver adds label `Product Management Approved` ([OpenLiberty/product-management](https://github.com/orgs/OpenLiberty/teams/product-management))
  - Note: For stabilized, superseded, and discontinued feature/capability, skip the [Beta](#beta) section of the template (you may delete it).  Otherwise, proceed as normal.

### **FAT Documentation**
- [x] "Feature Test Summary" child task created
  - Use the [Feature Test Summary Template](https://github.com/OpenLiberty/open-liberty/issues/new?assignees=&labels=Feature+Test+Summary&template=feature_test_summary.md&title=)
  - Add FTS issue link to the [Documents](#documents) section.

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## **Implementation**

A feature must be [prioritized](https://github.com/orgs/OpenLiberty/projects/2) before any implementation work may begin to be delivered (inaccessible/no-ship).  However, a design focused approach should still be applied to features, and developers should think about the feature design prior to writing and delivering any code.  
Besides being prioritized, a feature must also be socialized (or No Design Approved) before any beta code may be delivered.  All new Liberty content must be inaccessible in our GA releases until it is [Feature Complete](#feature-complete) by either marking it `kind=noship` or [beta fencing](#beta-code) it.  
Code may not GA until this feature has obtained the `Design Approved` or `No Design Approved` label, along with all other tasks outlined in the [GA](#ga) section.

### **Feature Development Begins**
- [x] Add the `In Progress` label

## **Legal and Translation**

In order to avoid last minute blockers and significant disruptions to the feature, the **legal items need to be done as early in the feature process as possible**, either in design or as early into the development as possible.  Similarly, translation is to be done concurrently with development.  All items below **MUST** be completed before beta & GA is requested.

### **Innovation** (Complete 1 week before Beta & GA Feature Complete Date)
- [x] Consider whether any aspects of the feature may be patentable. If any identified, disclosures have been submitted.

### **Legal** (Complete before Beta & GA Feature Complete Date)
- [x] Changed or new open source libraries are cleared and approved, or N/A. (Legal Release Services/Cass Tucker/Release PM).

### **Translation** (Complete by Beta & GA Feature Complete Date)
- [x] PII (Program Integrated Information) updates are merged (i.e. all English strings due for translation have been delivered), or N/A.

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## **Beta**

In order to facilitate early feedback from users, all new features and functionality should first be released as part of a beta release.

### **Beta Code**
- [x] [Beta fence](https://github.com/OpenLiberty/open-liberty/wiki/Beta-Fencing) the functionality
  - E.g. `kind=beta`, `ibm:beta`, `ProductInfo.getBetaEdition()`
- [x] Beta development complete and feature ready for inclusion in a beta release
  - Add label `target:beta` and the appropriate `target:YY00X-beta` (where YY00X is the targeted beta version) to the feature issue.
    - Note: This is expected to be done only once, for the initial beta that includes this feature. You do not need to add a `target:YY00(X+1)-beta`, `target:YY00(X+2)-beta`, etc. label for each additional beta that includes this feature.
- [x] Feature delivered into beta
  - ([OpenLiberty/release-manager](https://github.com/orgs/OpenLiberty/teams/release-manager)) adds label `release:YY00X-beta` (where YY00X is the first beta version that included the functionality).

### **Beta Blog** (Complete by beta eGA)
- [x] Beta blog issue created and populated using the [Open Liberty BETA blog post](https://github.com/OpenLiberty/open-liberty/issues/new/choose) template.
  - Add a link to the beta blog issue in the [Documents](#documents) section.
  - Note: This is for inclusion into the overall [beta release blog post](https://openliberty.io/blog/?search=beta&key=tag).  If, in addition, you'd also like to create a dedicated blog post about your feature, then follow the "Standalone Feature Blog Post" instructions under the [Other Deliverables](#other-deliverables) section.
  - A feature may have multiple beta blogs associated with it. This is especially useful for features that are continuously adding functionality each release and want to advertise what is new since the previous beta.
    - Each beta blog issue should have the appropriate `target:YY00X-beta` label added to it.
    - Include each beta blog issue in the [Documents](#documents) section.

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## **GA**

A feature is ready to GA after it is Feature Complete and has obtained all necessary Focal Point Approvals. 

### **Feature Complete**
- [x] Feature implementation and tests completed.
  - [x] All PRs are merged.
  - [x] All related/child issues are closed.
  - [x] All stop ship issues are completed.
- [ ] Legal: all necessary approvals granted.
- [ ] Innovation: IP identified and any applicable disclosures submitted
- [ ] Translation: Feature may only proceed to GA if it has either `Translation - Not Required`, `Translation - Complete`, or `Translation - Missing` label
  - If the feature does not have anything that required translation, the feature owner adds the label `Translation - Not Required`.
  - If all translation has been delivered to `release` branch, feature owner adds label `Translation - Complete`.
  - If missing translation does not cause a break in functionality, nor a security or production outage risk, feature owner adds label `Translation - Missing`.
    - Once all missing translations are delivered, the `Translation - Missing` label is replaced with `Translation - Complete`.
  - If missing translation could cause a break in functionality or a security or production outage risk, feature owner adds the `Translation - Blocked` label.
    - Features with `Translation - Blocked` may **NOT** proceed to GA until the label has been replaced with either `Translation - Missing` or `Translation - Complete`.
  - For further guidance, contact [Globalization focal point](https://github.com/orgs/OpenLiberty/teams/globalization-approvers) or the [Release Architect](https://github.com/orgs/OpenLiberty/teams/release-architect).
- [ ] GA development complete and feature ready for inclusion in a GA release
  - Add label `target:ga` and the appropriate `target:YY00X` (where YY00X is the targeted GA version).
  - Inclusion in a release requires the completion of all Focal Point Approvals.

### **Focal Point Approvals** (Complete by Feature Complete Date)

These occur only after GA of this feature is requested (by adding a `target:ga` label).  GA of this feature may not occur until all approvals are obtained.

### **All Features**
- [x] **APIs/Externals** - Externals have been reviewed or N/A.  ([OpenLiberty/externals-approvers](https://github.com/orgs/OpenLiberty/teams/externals-approvers))
  - Approver adds label `focalApproved:externals` 
- [x] **Demo** - Demo is scheduled for an upcoming EOI or N/A. ([OpenLiberty/demo-approvers](https://github.com/orgs/OpenLiberty/teams/demo-approvers))
  - Add comment `@OpenLiberty/demo-approvers Demo scheduled for EOI [Iteration Number]` to this issue.
  - Approver adds label `focalApproved:demo`.
- [x] **FAT** - All Tests complete and running successfully in SOE or N/A. ([OpenLiberty/fat-approvers](https://github.com/orgs/OpenLiberty/teams/fat-approvers))
  - Approver adds label `focalApproved:fat`.

### **Design Approved Features**
- [x] **ID** - Documentation is complete or N/A. ([OpenLiberty/id-approvers](https://github.com/orgs/OpenLiberty/teams/id-approvers))
  - Approver adds label `focalApproved:id`.
  - > **_NOTE:_**  If only trivial documentation changes are required, you may reach out to the ID Feature Focal to request a `ID Required - Trivial` label.  Unlike features with regular ID requirement, those with `ID Required - Trivial` label do not have a hard requirement for a Design/UFO.
- [x] **InstantOn** - InstantOn capable or N/A. ([OpenLiberty/instantOn-approvers](https://github.com/orgs/OpenLiberty/teams/instantOn-approvers))
  - Approver adds label `focalApproved:instantOn`.
- [x] **Performance** - Performance testing is complete or N/A. ([OpenLiberty/performance-approvers](https://github.com/orgs/OpenLiberty/teams/performance-approvers))
  - Approver adds label `focalApproved:performance`.
- [x] **Serviceability** - Serviceability has been addressed or N/A. ([OpenLiberty/serviceability-approvers](https://github.com/orgs/OpenLiberty/teams/serviceability-approvers))
  - Approver adds label `focalApproved:sve`.
- [x] **STE** - Skills Transfer Education chart deck is complete or N/A. ([OpenLiberty/ste-approvers](https://github.com/orgs/OpenLiberty/teams/ste-approvers))
  - Approver adds label `focalApproved:ste`.
- [x] **SVT** - System Verification Test is complete or N/A. ([OpenLiberty/svt-approvers](https://github.com/orgs/OpenLiberty/teams/svt-approvers))
  - Approver adds label `focalApproved:svt`.

### **Remove Beta Fencing** (Complete by Feature Complete Date)
- [x] Beta guards are removed, or N/A
  - Only after all necessary Focal Point Approvals have been granted.

### **GA Blog** (Complete by Friday after GM)
- [ ] GA Blog issue created and populated using the [Open Liberty GA release blog post](https://github.com/OpenLiberty/open-liberty/issues/new/choose) template.
  - Add a link to the GA Blog issue in the [Documents](#documents) section.
  - Note: This is for inclusion into the overall [release blog post](https://openliberty.io/blog/?search=release&key=tag).  If, in addition, you'd also like to create a dedicated blog post about your feature, then follow the "Standalone Feature Blog Post" instructions under the [Other Deliverables](#other-deliverables) section.

### **Post GM** (Complete before GA)
- [ ] After confirming this feature has been included in the GM driver, feature owner closes this issue.

### **Post GA**
- [ ] Remove the `target:ga` and `target:YY00X` labels, and add the appropriate `release:YY00X`. ([OpenLiberty/release-manager](https://github.com/orgs/OpenLiberty/teams/release-manager))

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## **Other Deliverables**

- [ ] **Standalone Feature Blog Post** - A blog post specifically about your feature or N/A.  (Feature owner and [OpenLiberty/release-architect](https://github.com/orgs/OpenLiberty/teams/release-architect))
  - This should be strongly considered for larger or more prominent features.
  - Follow [instructions](https://github.com/OpenLiberty/blogs/tree/draft#writing-and-publishing-blog-posts-on-the-openlibertyio-blog) in the blogs repo.
- [ ] **OL Guides** - OL Guides assessment is complete or N/A. ([OpenLiberty/guide-assessment](https://github.com/orgs/OpenLiberty/teams/guide-assessment/members))
- [ ] **Dev Experience** - Developer Experience & Tools work is complete or N/A. ([OpenLiberty/dev-experience-assessment](https://github.com/orgs/OpenLiberty/teams/dev-experience-assessment/members))
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

**Comment by idlewis** [comment]
Comments from the UFO Socialization:
Page 11: The 'first line of request' attribute should be removed as it is effectively duplicated by the 'body' attribute
Page 12: The last attribute description is incorrect
Page 13: Check that this matches the current set of attributes from pages 11/12
Page 30: Performance testing should compare 'just http access logs' with 'http access logs + open telemetry'
Page 33: Will there be a message to indicate that this feature is active?

@tonyreigns

**Comment by donbourne** [comment]
@tonyreigns , please add the "Design Approval Request" label when you are ready. Also, please be sure the UFO is posted and linked from this issue and shared before adding that.

**Comment by NottyCode** [comment]
@tonyreigns On slide 19 it introduces `accessLogFields` but I don't understand what it does. It says it can be set to `logFormat` or `default` but I don't understand what they mean and what effect it has.

**Comment by tonyreigns** [comment]
Hi @NottyCode , these values and attributes are being copied over from JSON Logging and Log Stash Collector. We're just adding Telemetry support so I didn't expand much on it. This is not new functionality that we came up with but rather we just ported it over - with the attribute name `accessLogFields` instead of `jsonAccessLogFields` being the only difference.

With access logging, you can either configure a custom collection of information to include in each access log, which you would name the id `logFormat` or you can specify `default` to use a pre-selected collection of information.

So for the example I gave,
`<httpAccessLogging id="accessLogging" logFormat='%h %u’/>`, the custom config here will only include the remote host and remote user.

Then when you specify `accessLogFields=“logFormat”`, you're configuring all telemetry logs that are bridged to OTEL to be the above two pieces of information. If you set it to default, you'll use the same default configuration for JSON Logging and LogStash.

I updated the UFO on slides 19 and 20 to include this information. Let me know if you need further clarification. Thanks.

**Comment by NottyCode** [comment]
@tonyreigns where does `jsonAccessLogFields` come from? It doesn't appear to be on `httpAcessLogging` or on `mpTelemetry` or `logging`.

Slide 20 says `accessLogFields` references `accessLogging` on `httpEndpoint`, but I think you mean `httpAccessLogging`

Slide 20 doesn't say what fields are selected if default it set and it should. Are you using an enum for the values?

**Comment by tonyreigns** [comment]
@NottyCode 

`jsonAccessLogFields` is referenced in the jsonLogging and logstash documentation: 
JSON Logging: https://openliberty.io/docs/latest/log-trace-configuration.html#http
LogStash Collector: https://openliberty.io/docs/latest/reference/feature/logstashCollector-1.0.html#_replace_default_http_access_log_fields

For Telemetry, jsonAccessLogFields didn't make much sense so the new attribute is "accessLogFields" which will only be applicable for the mpTelemetry element:
`<mpTelemetry accessLogFields=“logFormat” source=“message,trace,ffdc,accessLog”/>`

For slide 20, I'm pointing out that inside the `httpEndpoint` element, you need to reference the `httpAccessLogging` ID that you configured previously. I'll reword this to make it more clear.

I don't think we have the default attributes documented, so I'll collect them and update the UFO shortly. We're piggybacking off the jsonLogging functionality and they're not using enum for the values, they have their own fieldSetter logic.

**Comment by NottyCode** [comment]
@tonyreigns I'm going to approve, but I think we need to do a better job on documenting `accessLogFields`. Neither the generated documentation of the log trace configuration document explain what the `default` fields are.

**Comment by tonyreigns** [comment]
Thank you and I agree, I previously thought it was documented. I'll create an issue to update the appropriate docs.

**Comment by tonyreigns** [comment]
@OpenLiberty/externals-approvers  Can you please review the approval for this feature, there are no exposed public APIs as part of this particular feature. Thanks.

**Comment by tonyreigns** [comment]
@OpenLiberty/demo-approvers  Demo scheduled for EOI 25.06

Thanks.

**Comment by tonyreigns** [comment]
@OpenLiberty/id-approvers  ID Doc Issue opened: https://github.com/OpenLiberty/docs/issues/7872

Thanks.

**Comment by dave-waddling** [comment]
@tonyreigns: Just after granting the FAT approval, I spotted that new code for this feature was merged via #30893. While the change is simple, the approval requires all product and test code to have been validated by the SOE. I'll remove the approval and add it after the next SOE run against a driver which contains the PR.

**Comment by tonyreigns** [comment]
@dave-waddling I completely understand thank you!

**Comment by tonyreigns** [comment]
**UFO -- does the UFO identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR? Have these issues been addressed in the implementation?**

Yes, the UFO identifies the most likely problems customers will see. Below are the common problems that users may encounter:

1. No access logs are exported to OpenTelemetry: If access logging isn't configured or the access log source isn't specified, there wil lbe no access logs exported. These are documented and appropriate messages will be also logged to let customers know.
2. Invalid log source name for accessLog: A warning message will be logged.
3. accessLogFields set to an invalid value. A warning message will be logged and the accessLogFields attribute will be set to default automatically.


**Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. IBM Support, test team, or another development team).**
a) What problem paths were tested and demonstrated?
    The problems paths mentioned above were also demoed to analyze the problem paths.
b) Who did you demo to?
    Local dev team, SVT, and this will be presented in the next EOI.
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that IBM Support should be able to quickly address those problems without need to engage SMEs?
    Yes.

**SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them. Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.**
a) Who conducted SVT tests for this feature?
    Brian Hanczaryk 
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that IBM Support should be able to quickly address those problems without need to engage SMEs?
    Yes, SVT agrees that the serviceability of any problem encountered was sufficient to avoid PMRs or L2 should be able to quickly address those problems without engaging L3.


**Which IBM Support / SME queues will handle PMRs for this feature? Ensure they are present in the contact reference file and in the queue contact summary, and that the respective IBM Support/SME teams know they are supporting it. Ask Don Bourne if you need links or more info.**
    WAS L3: Logging

**Does this feature add any new metrics or emit any new JSON events? If yes, have you updated the JMX metrics reference list / Metrics reference list / JSON log events reference list in the Open Liberty docs?**
    No.

**Comment by tngiang73** [comment]
@tonyreigns : WASWIN is good with the STE slides. Hence, approving the STE.

**Comment by dave-waddling** [comment]
@tonyreigns: Thanks, I appreciate the understanding! Unfortunately, #30893 didn't make it into a Release build in time for the SOE that launched this morning. The next run will be launched on 2025-04-04 and complete on 2025-04-07, which is the day before the 25.0.0.4 GM and so very tight to be adding/removing beta guards depending on approval. I'll reach out on Slack to discuss further.

**Comment by dave-waddling** [comment]
@tonyreigns: Thanks for the chat on Slack, very helpful! I agree that the risk posed by the changes in #30893 is low, and given that the feature can be confidently validated with just two FATs a mini-SOE next week is containable. As you're adding some additional test cases, please can you link the PR(s) here so that I can follow them through into a Release build. Then, early next week, I'll get an SOE run kicked off that runs the following FATs:
* `io.openliberty.microprofile.telemetry.2.0.logging.internal_2_fat` 
* `io.openliberty.microprofile.telemetry.2.0.logging.internal.container_fat`

I'll let you know when it's launched!

**Comment by tonyreigns** [comment]
Thanks @dave-waddling 

The PR with the additional testing and error handling is here: https://github.com/OpenLiberty/open-liberty/pull/31118 

These changes include:

- Additional error handling for an edge case that will never happen naturally, but would fail if you purposefully manually passed invalid data(these headers are automatically generated). Debug messages are now logged in this case.

4 tests to `io.openliberty.microprofile.telemetry.2.0.logging.internal_2_fat`:

- 3 testing successful trace parsing with different propagation configuration and 1 invalid trace(edge case coverage). 

1 test to `io.openliberty.microprofile.telemetry.2.0.logging.internal.container_fat`:

- Verify successful access log bridging to an otlp container.

Appreciated!

**Comment by dave-waddling** [comment]
@tonyreigns: FYI your change made it into a Release build and I've now kicked off the run. It'll be bumping up against the tail end of the full weekend SOE run initially, but should complete before EOD tomorrow.

**Comment by tonyreigns** [comment]
Thanks Dave! While the mini-SOE looks good, the team decided to push this feature to 25005 in order to add support for some additional user configuration. A small code change(with an accompanying test) will be added in a week or so. Sorry about the trouble.

**Comment by chirp1** [comment]
Isaac is working on the documentation for this issue. He should be able to complete the doc for 25.0.0.5. The doc issue is at https://github.com/OpenLiberty/docs/issues/7872. Approving this feature.

**Comment by tonyreigns** [comment]
All approvals have been received and beta guard has been removed.

**Comment by tonyreigns** [comment]
This feature was included and released in the 25.0.0.5 GA. Closing the issue now.
